acceptance/internal/prepare_server.go

```diff
@@ -35,78 +38,108 @@ func isTruePtr(value *bool) bool {
 	return value != nil && *value
 }
 
-func PrepareServerAndClient(t *testing.T, config TestConfig, logRequests bool, outputDir string) *databricks.WorkspaceClient {
+// This function returns the workspace auth configuration that should be used to authenticate
+// the "script" that's executed as part of the acceptance test.
+func PrepareServerAndClient(t *testing.T, config TestConfig, logRequests bool, outputDir string) (*sdkconfig.Config, iam.User) {
 	cloudEnv := os.Getenv("CLOUD_ENV")
+	recordRequests := isTruePtr(config.RecordRequests)
+
+	// If we are running in a cloud environment AND we are recording requests,
+	// start a dedicated server to act as a reverse proxy to a real Databricks workspace.
+	if cloudEnv != "" && recordRequests {
+		// Use a non-proxy client to fetch user info so that this API call is not recorded
+		// in out.requests.txt
+		w, err := databricks.NewWorkspaceClient()
+		require.NoError(t, err)
+
+		user, err := w.CurrentUser.Me(context.Background())
+		require.NoError(t, err, "Failed to get current user")
```

Comment:
```
		w, err := databricks.NewWorkspaceClient()
		require.NoError(t, err)

		user, err := w.CurrentUser.Me(context.Background())
		require.NoError(t, err, "Failed to get current user")
```

This section is always executed when cloudEnv != "" but it is now duplicated. Can we extract it in a separate shared block?

Reply:
Good point, done.